pkg/controller/command/store/command.go

```diff
@@ -0,0 +1,188 @@
+	DeleteErrorCode
+)
+
+var logger = log.New("agent-sdk-store")
+
+// Provider describes dependencies for the client.
+type Provider interface {
+	StorageProvider() storage.Provider
+}
+
+// Command is controller command for store.
+type Command struct {
+	store storage.Store
+}
+
+// New returns new store controller command instance.
+func New(p Provider) (*Command, error) {
+	store, err := p.StorageProvider().OpenStore(CommandName)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Command{store: store}, nil
+}
+
+// GetHandlers returns list of all commands supported by this controller command.
+func (c *Command) GetHandlers() []command.Handler {
+	return []command.Handler{
+		cmdutil.NewCommandHandler(CommandName, GetCommandMethod, c.Get),
+		cmdutil.NewCommandHandler(CommandName, PutCommandMethod, c.Put),
+		cmdutil.NewCommandHandler(CommandName, IteratorCommandMethod, c.Iterator),
+		cmdutil.NewCommandHandler(CommandName, DeleteCommandMethod, c.Delete),
+	}
+}
+
+// Get fetches the record based on key.
+func (c *Command) Get(rw io.Writer, req io.Reader) command.Error {
+	var request GetRequest
+
+	err := json.NewDecoder(req).Decode(&request)
+	if err != nil {
+		logutil.LogError(logger, CommandName, GetCommandMethod, err.Error())
+
+		return command.NewValidationError(InvalidRequestErrorCode, err)
+	}
+
+	result, err := c.store.Get(request.Key)
+	if err != nil {
+		logutil.LogError(logger, CommandName, GetCommandMethod, err.Error())
+
+		return command.NewExecuteError(GetErrorCode, err)
+	}
+
+	command.WriteNillableResponse(rw, &GetResponse{
+		Result: result,
+	}, logger)
+
+	logutil.LogDebug(logger, CommandName, GetCommandMethod, successString)
+
+	return nil
+}
+
+// Put stores the key and the record.
+func (c *Command) Put(rw io.Writer, req io.Reader) command.Error {
+	var request PutRequest
+
+	err := json.NewDecoder(req).Decode(&request)
+	if err != nil {
+		logutil.LogError(logger, CommandName, PutCommandMethod, err.Error())
+
+		return command.NewValidationError(InvalidRequestErrorCode, err)
+	}
+
+	if err = c.store.Put(request.Key, request.Value); err != nil {
+		logutil.LogError(logger, CommandName, PutCommandMethod, err.Error())
+
+		return command.NewExecuteError(PutErrorCode, err)
+	}
+
+	command.WriteNillableResponse(rw, nil, logger)
+
+	logutil.LogDebug(logger, CommandName, PutCommandMethod, successString)
+
+	return nil
+}
+
+// Iterator retrieves data according to given start and end keys.
+func (c *Command) Iterator(rw io.Writer, req io.Reader) command.Error {
+	var request IteratorRequest
+
+	err := json.NewDecoder(req).Decode(&request)
+	if err != nil {
+		logutil.LogError(logger, CommandName, IteratorCommandMethod, err.Error())
+
+		return command.NewValidationError(InvalidRequestErrorCode, err)
+	}
+
+	records := c.store.Iterator(request.StartKey, request.EndKey)
+	defer records.Release()
+
+	var values [][]byte
+
+	for records.Next() {
+		values = append(values, records.Value())
```

Comment:
this will load all records in memory.

Reply:
I know :( But I need this at least for this release.